roles/ami_config/tasks/rh_subscription.yml

```diff
@@ -0,0 +1,23 @@
+---
+# Subscribe the AMIs to RH
+
+- name: Register as user
+  redhat_subscription:
+    state: present
+    username: "{{ rhn_username }}"
+    password: "{{ rhn_password }}"
+    pool_ids:
```

Comment:
Instead of specifying multiple pools as different variables, I would just reference a single variable which is a list

```
pool_ids: "{{ rhn_pool_ids }}"
```